tasks/CommandInjection.cpp

```diff
@@ -0,0 +1,83 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "CommandInjection.hpp"
+
+using namespace auv_control;
+
+CommandInjection::CommandInjection(std::string const& name)
+    : CommandInjectionBase(name)
+{
+}
+
+CommandInjection::CommandInjection(std::string const& name, RTT::ExecutionEngine* engine)
+    : CommandInjectionBase(name, engine)
+{
+}
+
+CommandInjection::~CommandInjection()
+{
+}
+
+bool CommandInjection::calcOutput()
+{
+    base::LinearAngular6DCommand output_command = merged_command;
+
+    if(_cmd_injection.connected())
+    {
+        base::LinearAngular6DCommand cmd_injection;
+        if(_cmd_injection.readNewest(cmd_injection) == RTT::NewData)
```

Comment:
Oh ... and don't forget to reinitialize newest_injection_sample in the startHook